packages/components/src/components/trace/TraceFilter.vue

```diff
@@ -0,0 +1,354 @@
+          <b>{{ currentIndex + 1 }}</b>
+          /
+        </span>
+        {{ nodesLength }} results
+      </div>
+      <div class="trace-filter__arrow" @click="onNextArrow">
+        <ArrowSearchRightIcon />
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import ArrowSearchLeftIcon from '@/assets/arrow-search-left.svg';
+import ArrowSearchRightIcon from '@/assets/arrow-search-right.svg';
+import CloseThinIcon from '@/assets/close-thin.svg';
+import SearchIcon from '@/assets/search.svg';
+import { getEventTarget } from '@appland/diagrams';
+
+export default {
+  name: 'v-trace-filter',
+
+  components: {
+    ArrowSearchLeftIcon,
+    ArrowSearchRightIcon,
+    CloseThinIcon,
+    SearchIcon,
+  },
+
+  props: {
+    nodesLength: {
+      type: Number,
+      require: false,
+      default: 0,
+    },
+    currentIndex: {
+      type: Number,
+      default: null,
+    },
+    suggestions: {
+      type: Array,
+      default: () => [],
+    },
+  },
+
+  data() {
+    return {
+      filterValue: '',
+      showSuggestions: false,
+      selectedSuggestion: 0,
+    };
+  },
+
+  watch: {
+    filterValue: {
+      handler(value) {
+        this.$emit('onChange', value);
+      },
+    },
+    selectedSuggestion: {
+      handler(selectedSuggestion) {
+        const selected =
+          this.$refs.suggestionsList.querySelectorAll('li')[selectedSuggestion];
+
+        if (selected.offsetTop < this.$refs.suggestionsList.scrollTop) {
+          this.$refs.suggestionsList.scrollTop = selected.offsetTop;
+        } else if (
+          selected.offsetTop >=
+          this.$refs.suggestionsList.scrollTop +
+            this.$refs.suggestionsList.offsetHeight
+        ) {
+          this.$refs.suggestionsList.scrollTop =
+            selected.offsetTop - selected.offsetHeight * 2;
+        }
+      },
+    },
+  },
+
+  computed: {
+    formClasses() {
+      const classList = ['trace-filter__form'];
+
+      if (this.showSuggestions && this.suggestions.length) {
+        classList.push('trace-filter__form--show-suggestions');
+      }
+
+      return classList;
+    },
+  },
+
+  methods: {
+    setValue(newValue) {
+      this.filterValue = newValue;
+    },
+    clearValue() {
+      this.filterValue = '';
+    },
+    onPrevArrow() {
+      this.$emit('onPrevArrow');
+    },
+    onNextArrow() {
+      this.$emit('onNextArrow');
+    },
+    getSuggestionClasses(index) {
+      const classes = ['trace-filter__suggestions-item'];
+      if (index === this.selectedSuggestion) {
+        classes.push('trace-filter__suggestions-item--selected');
+      }
+      return classes;
+    },
+    onInputFocus() {
+      this.showSuggestions = true;
+      this.selectedSuggestion = 0;
+    },
+    onFormSubmit() {
+      if (
+        this.$refs.input === window.document.activeElement &&
+        this.showSuggestions &&
+        this.suggestions.length
+      ) {
+        this.makeSelection(this.suggestions[this.selectedSuggestion].id);
```

Comment:
I'd expect this to input the same text that's suggested, meaning we're searching by name and not an identifier. The suggestions would need to be de-duplicated as a result.

![image](https://user-images.githubusercontent.com/8737782/152193639-39549530-90d5-4399-9986-833aaf848feb.png)


https://user-images.githubusercontent.com/8737782/152193373-dfe9980c-43cb-40ce-a3d7-944747d25087.mp4

Reply:
This is somewhat complicated as a suggestion may contain spaces, and we have no way of filtering a literal that contains whitespace because we split the search term into queries delimited by whitespace. `POST /users` will search for `POST && /users`. Ideally we'd allow the query to be quoted (`"POST /users"`) and not process any whitespace contained within.

Reply:
Now picked event name will be present in filter input in double quotes https://github.com/applandinc/appmap-js/pull/534/commits/16d5e5d02a8ac984eb15a64001f3398079c10f74

Reply:
@dustinbyrne it's already implemented.
If you type "POST /users", filtering algorithm will look up for event with that name, here is the code which responsible for that:
https://github.com/applandinc/appmap-js/blob/feat/trace-filter-auto-complete/packages/components/src/pages/VsCodeExtension.vue#L616-L626